internal/locate/sorted_btree.go

```diff
@@ -80,24 +80,40 @@ func (s *SortedRegions) SearchByKey(key []byte, isEndKey bool) (r *Region) {
 }
 
 // AscendGreaterOrEqual returns all items that are greater than or equal to the key.
-// The first region's startKey is `startKey`, all regions returned are continuous and have no hole.
 func (s *SortedRegions) AscendGreaterOrEqual(startKey, endKey []byte, limit int) (regions []*Region) {
-	lastStartKey := startKey
 	s.b.AscendGreaterOrEqual(newBtreeSearchItem(startKey), func(item *btreeItem) bool {
 		region := item.cachedRegion
 		if len(endKey) > 0 && bytes.Compare(region.StartKey(), endKey) >= 0 {
 			return false
 		}
-		if !region.checkRegionCacheTTL(time.Now().Unix()) {
+		regions = append(regions, region)
+		return len(regions) < limit
+	})
+	return regions
+}
+
+// DescendLess returns all items that are less than or equal to the key.
+func (s *SortedRegions) DescendLessOrEqual(startKey, endKey []byte) (regions []*Region) {
```

Comment:
Another question, why or-equal for this location func? IMO, `endKey` should be excluded bound.

Reply:
BTree doesn't provide the function DescendLess...

Reply:
I'm a little concened here that this function holds a global lock, if it takes too long time, maybe this sql will affect the other sqls, especially the oltp queries.